apps/src/blockly/customBlocks/googleBlockly/mutators/behaviorDefMutator.js

```diff
@@ -73,7 +73,7 @@ export const behaviorDefMutator = {
         this.description = node.textContent;
       }
     }
-    this.behaviorId = xmlElement.nextElementSibling.getAttribute('id');
+    this.behaviorId = xmlElement.getAttribute('behaviorId');
```

Comment:
why don't we need to do anything with `userCreated` here?

Reply:
`userCreated` just exists to tell Blockly that it should update the `behaviorId` when the behavior gets renamed.

Reply:
Do we not need to preserve it when parsing xml?

Reply:
Oh, you're right, I think we do. Apologies for not understanding your original question. This would be needed to get the property onto the block when loading XML. I did most of my testing with JSON sources, so I had a gap here. Will update!